### PR TITLE
feat: plotnine confusion matrix

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Created a confusion matrix, and abstracted it into common.

@HLasse, please review the code in common, and feel free to add commons on the code in t2d :-)

For T2D, needed something that was a plotnine plot so I can use it with patchworklib.

Unfortunately, `matplotlib` does not allow moving the x-axis on the top of the plot, so quite limited in the options here.

Output:
image

### CODE REVIEW

File: src/psycop/common/model_evaluation/confusion_matrix/confusion_matrix.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    @property
+    def specificity(self) -> float:
+        return self.true_negatives / (self.true_negatives + self.false_positives)
+
+
+def get_confusion_matrix_cells_from_long_df(
+    long_df: pd.DataFrame,
+) -> ConfusionMatrix:
+    """Get confusion matrix cells from a long dataframe."""
```
Comment (HLasse) [inline src/psycop/common/model_evaluation/confusion_matrix/confusion_matrix.py:33]: Please add a comment on the required columns/format - long_df is a bit too unspecific. It also seems that the df should actually be wide with columns "true" and "pred"?

Comment (HLasse) [reply src/psycop/common/model_evaluation/confusion_matrix/confusion_matrix.py:33]: EDIT2, or not? - is `create_long_confusion_matrix_df` supposed to be called by the user? In that case I would move it away from `test_confusion_matrix.py` and rather have it in this file.

File: src/psycop/projects/t2d/paper_outputs/model_description/performance/plotnine_confusion_matrix.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Yes," ","Sens:\n{round(matrix.sensitivity, 2)}",
+""",
+    )
+
+    """Create a confusion matrix and return a plotnine object."""
+    df["true"] = pd.Categorical(df["true"], ["Yes", "No", " "])
+    df["pred"] = pd.Categorical(df["pred"], ["+", "-", " "])
+
+    p = (
+        pn.ggplot(df, pn.aes(x="true", y="pred", fill="estimate"))
```
Comment (HLasse) [inline src/psycop/projects/t2d/paper_outputs/model_description/performance/plotnine_confusion_matrix.py:29]:
```suggestion
        pn.ggplot(df, pn.aes(x="pred", y="true", fill="estimate"))
```
Predictions are usually the columns, and true values the rows (see e.g. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ConfusionMatrixDisplay.html#sklearn.metrics.ConfusionMatrixDisplay)

**Review by HLasse — APPROVED** [review]
Overall looks, although I'm a bit confused as to what interface the user meets. `get_confusion_matrix_cells_from_long_df` returns the object that's used for the plotting function, but it looks documentation on the expected input. `create_long_confusion_matrix_df` does not seem to be used for anything, consider removing

File: src/psycop/projects/t2d/paper_outputs/model_description/performance/plotnine_confusion_matrix.py (hunk 2 above)
Comment (MartinBernstorff) [reply src/psycop/projects/t2d/paper_outputs/model_description/performance/plotnine_confusion_matrix.py:29]: Seems that's quite inconsistent in general D:

https://www.google.com/search?q=confusion+matrix&tbm=isch&ved=2ahUKEwiSk93BjPz-AhUOwioKHQo_AOUQ2-cCegQIABAA&oq=confusion+matrix&gs_lcp=CgNpbWcQAzIECCMQJzIECCMQJzIFCAAQgAQyBQgAEIAEMgUIABCABDIFCAAQgAQyBQgAEIAEMgUIABCABDIFCAAQgAQyBQgAEIAEUJEHWJ4IYMoLaABwAHgAgAFNiAHRAZIBATOYAQCgAQGqAQtnd3Mtd2l6LWltZ8ABAQ&sclient=img&ei=JaZkZNLsNI6EqwGK_oCoDg&bih=2056&biw=1920

**Comment by MartinBernstorff** [comment]
Thanks a tonne, extremely helpful! I've updated the docs and removed the unused function.

File: src/psycop/projects/t2d/paper_outputs/model_description/performance/plotnine_confusion_matrix.py (hunk 2 above)
Comment (MartinBernstorff) [reply src/psycop/projects/t2d/paper_outputs/model_description/performance/plotnine_confusion_matrix.py:29]: Let me know if you have strong opinions on this, then I'll fix it for the manuscript.
